AdminUi/src/AdminUi/ClientApp/src/app/components/quotas/assign-quotas-dialog/assign-quotas-dialog.component.ts

```diff
@@ -62,6 +66,38 @@ export class AssignQuotasDialogComponent {
         });
     }
 
+    public createIdentityQuota(quotaData: AssignQuotaData): void {
+        this.loading = true;
+
+        const createQuotaRequest = {
+            metricKey: quotaData.metricKey,
+            max: quotaData.max,
+            period: quotaData.period
+        } as CreateQuotaForIdentityRequest;
+
+        this.quotasService.createIdentityQuota(createQuotaRequest, this.data.address).subscribe({
+            next: (_: HttpResponseEnvelope<IdentityQuota>) => {
+                this.quotasService.triggerRefresh();
+                this.isAllOk = true;
+                this.snackBar.open("Successfully assigned quota.", "Dismiss", {
+                    duration: 4000,
+                    verticalPosition: "top",
+                    horizontalPosition: "center"
+                });
+            },
+            complete: () => {
+                this.loading = false;
+                if (this.isAllOk) {
+                    this.dialogRef.close();
+                }
```

Comment:
Why not just move this to `next`? Then you can get rid of this weird `isAllOk` variable.